module2-database-technologies/README.md

```diff
@@ -1 +1,372 @@
+* A table has many rows (also known as records)
+* A row has many columns (also known as fields).
+
+For example, the previously mentioned `Customer` table may look something like
+the following. Note that this is not any particular syntax but simply a
+visualization of the data.
+
+**id**|**first\_name**|**last\_name**|**registered\_at**
+:-----:|:-----:|:-----:|:-----:
+1|Ammar|Sammour|2012-01-02
+2|Halit|Batur|2012-01-03
+3|Shrreya|Bhatachaarya|2012-01-05
+
+The `Order` table may look something like so:
+
+**id**|**product**|**delivered**|**customer\_id**
+:-----:|:-----:|:-----:|:-----:
+1|Keyboard|TRUE|1
+2|Mouse|FALSE|1
+3|Cookies|TRUE|1
+4|Rice|TRUE|2
+
+On the `Order` table, note that there is a field called `customer_id`. This is
+called a **foreign key** (or sometimes join key). This column creates a
+**relation** between the `Customer` and `Order` tables: it tells us to which
+customer the order belongs. This is the core of relational databases: expressing
+relations between entities.
+
+This table says that Ammar has ordered a keyboard, mouse, and cookies. He has
+three orders, because there are three rows with `customer_id = 1`. Halit has ordered one item: rice (`customer_id = 2`).
+
+### Describing relationships
+We usually use the terms one-to-many and many-to-many to describe
+relationships between tables. 
+
+For example, a customer may have multiple orders. This is a one-to-many relationship,
+since one customer has many orders. It should be noted that the foreign key goes
+on the side that is "many". In the example above, note that the `customer_id`
+key is located on the `Order` table.
+
+For a many-to-many relationship, consider students and courses at a university.
+A student can take many courses; a course also has many students. Many-to-many
+relationships are [expressed slightly
+differently](https://dzone.com/articles/how-to-handle-a-many-to-many-relationship-in-datab) than using a single foreign key.
+
+See this [StackOverflow
+question](https://stackoverflow.com/questions/4601703/difference-between-one-to-many-and-many-to-one-relationship)
+for more details.
+
+### SQL queries
+
+Now that we understand the general concept behind relational data, how do we
+query it? Note that every SQL database may have slightly different syntax in
+terms of quotations, capitalization, commas, etc., so be sure to check. In these
+examples, we will be using MySQL.
+
+Suppose you have an existing MySQL database. To create the `customer` table (we
+will now lowercase the table names by convention), you would use the `CREATE
+TABLE` statement. Note that every column has a type.
+[`AUTO_INCREMENT`](https://www.guru99.com/auto-increment.html) is used to
+automatically generated the IDs. 
+
+```sql
+CREATE TABLE IF NOT EXISTS customer (
+    task_id INT AUTO_INCREMENT PRIMARY KEY,
+    first_name VARCHAR(255) NOT NULL,
+    last_name VARCHAR(255) NOT NULL,
+    created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP
+)  
+```
```

Comment:
I think we should add 1-2 lines explaining the anatomy of this query. Specially the significance of NOT NULL and DEFAULT. Even the line on AUTO_INCREMENT above can be moved here instead. And the usage of PRIMARY KEY and UNIQUE KEY.
And then we can link the MySQL documentation page link for CREATE TABLE to suggest exploring the query syntax and options in detail.